src/libYARP_sig/include/yarp/sig/PointCloudNetworkHeader.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2006-2018 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+#ifndef YARP_SIG_POINTCLOUDNETWORKHEADER_H
+#define YARP_SIG_POINTCLOUDNETWORKHEADER_H
+
+#include <yarp/conf/system.h>
+
+#include <yarp/os/NetInt32.h>
+
+#include <yarp/sig/PointCloudTypes.h>
+
+namespace yarp {
+    namespace sig {
+        class PointCloudNetworkHeader;
+    }
+}
+
+
+// YARP_BEGIN_PACK
+/**
+ * @brief The yarp::sig::PointCloudNetworkHeader class
+ */
+class yarp::sig::PointCloudNetworkHeader
+{
+public:
+    PointCloudNetworkHeader() : width(0),
+                                height(0),
+                                pointType(0),
+                                isDense(true),
+//                                pose(4, 0.0f),
+//                                orientation(4, 0.0f),
+                                hasCustomData(false),
+                                pad{0,0}
+    { YARP_UNUSED(pad); }
+
+    // PCL like fields
+    yarp::os::NetInt32  width;
+    yarp::os::NetInt32  height;
+    yarp::os::NetInt32  pointType;       // bitwise of all possible informations -> could also be int64 or just an enum, but I thin bitwise gives more freedom about all possible combinations
+    bool                isDense;         // the point cloud is dense if not contains NaN or Inf values
+//     yarp::sig::Vector   pose;            // translation from origin -- could be an Eigen::Vector4f for better PCL compatibility if yarp can afford to depend from it
+//     yarp::sig::Vector   orientation;     // orientation wrt origin  -- could be an Eigen::Quaternion for better PCL compatibility if yarp can afford to depend from it
+
+    // YARPish fileds
+//     char                *data;           // actual pointCloud data.
```

Comment:
If all this commented line are not necessary, can we remove them?

Reply:
I think that it is better to keep it for the future improvements